File: manual/data/python/src/function_sortedcontainers_SortedList_bisect_left.py
from sortedcontainers import SortedList

class HanSeok:
    def find_insert_position(self, value: int = 0, sorted_list: str = None) -> int:
        if sorted_list is None:
            sorted_list = "1,3,5,8"
        sorted_list = sorted_list.split(",")
        sorted_list = [int(x) for x in sorted_list]
        sorted_list = SortedList(sorted_list)
        position = sorted_list.bisect_left(value)
        return position

File: code/generate_strategies/IR(pseudocode)/java/java/function_sortedcontainers_SortedList_bisect_left.java
package com.example;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import org.apache.commons.lang3.math.NumberUtils; 

public class HanSeok {
    
    public int find_insert_position(int value, String sorted_list) {
        
        if (sorted_list == null) {
            sorted_list = "1,3,5,8";
        }
        
        String[] parts = sorted_list.split(",");
        
        List<Integer> list = new ArrayList<>();
        for (String part : parts) {
            
            list.add(NumberUtils.toInt(part.trim()));
        }
        
        Collections.sort(list);
        
        int index = Collections.binarySearch(list, value);
        if (index < 0) {
            
            index = -index - 1;
        } else {
            
            while (index > 0 && list.get(index - 1) == value) {
                index--;
            }
        }
        return index;
    }
}